utils/host/src/hosts/eigenda.rs

```diff
@@ -0,0 +1,93 @@
+use std::sync::Arc;
+
+use alloy_eips::BlockId;
+use alloy_primitives::B256;
+use async_trait::async_trait;
+use kona_preimage::BidirectionalChannel;
+use op_succinct_client_utils::witness::WitnessData;
+
+use crate::{fetcher::OPSuccinctDataFetcher, hosts::OPSuccinctHost};
+use anyhow::Result;
+
+use hokulea_host_bin::cfg::SingleChainHostWithEigenDA;
+
+use kona_preimage::{HintWriter, NativeChannel, OracleReader};
+
+use crate::witness_generation::witness_generator::{EigenDAWitnessGenerator, WitnessGenerator};
+use hokulea_proof::eigenda_provider::OracleEigenDAProvider;
+
+use kona_proof::{l1::OracleBlobProvider, CachingOracle};
+
+#[derive(Clone)]
+pub struct EigenDAOPSuccinctHost {
+    pub fetcher: Arc<OPSuccinctDataFetcher>,
+    pub witness_generator: Arc<EigenDAWitnessGenerator>,
+}
+
+#[async_trait]
+impl OPSuccinctHost for EigenDAOPSuccinctHost {
+    type Args = SingleChainHostWithEigenDA;
+    type WitnessGenerator = EigenDAWitnessGenerator;
+
+    fn witness_generator(&self) -> &Self::WitnessGenerator {
+        &self.witness_generator
+    }
+
+    /// Run the host and client program.
+    ///
+    /// Returns the witness which can be supplied to the zkVM.
+    async fn run(&self, args: &Self::Args) -> Result<WitnessData> {
+        let hint = BidirectionalChannel::new()?;
+        let preimage = BidirectionalChannel::new()?;
+
+        let server_task = args.start_server(hint.host, preimage.host).await?;
+
+        let witness =
+            self.witness_generator().run_witnessgen_client(preimage.client, hint.client).await?;
+        // Unlike the upstream, manually abort the server task, as it will hang if you wait for both
+        // tasks to complete.
+        server_task.abort();
+
+        Ok(witness)
+    }
```

Comment:
Refac so that it lives inside the trait definition as a default implementation since they're all the same by witness_generator().run_witnessgen_client.

Reply:
Need trait for Args to let the compiler know that start_server is implemented for Args.

Having a trait in op-succinct to make this refac happen is not right IMO.

Asked for introducing a `Host` trait in [op r&d discord](https://discord.com/channels/1244729134312198194/1244729507588608040/1364693480575598743).